spec/unit/installer/xcode/multi_pods_project_generator_spec.rb

```diff
@@ -723,22 +740,58 @@ class Xcode
               ]
               test_scheme.test_action.code_coverage_enabled?.should.be.true
+              test_scheme.launch_action.macro_expansions.empty?.should.be.false
+            end
+
+            it 'adds the test bundle to the test action of the app host when using app specs' do
+              @generator.installation_options.
+                  stubs(:share_schemes_for_development_pods).
+                  returns(true)
+              @generator.sandbox.stubs(:development_pods).returns('PineappleLib' => fixture('pineapple-lib'))
+
+              pod_generator_result = @generator.generate!
+
+              projects_by_pod_targets = pod_generator_result.projects_by_pod_targets
+              pineapple_project = projects_by_pod_targets.keys.find { |p| p.project_name == 'PineappleLib' }
+
+              Xcode::PodsProjectWriter.new(config.sandbox, [pineapple_project],
+                                           pod_generator_result.target_installation_results.pod_target_installation_results,
+                                           @generator.installation_options).write!
+
+              @generator.configure_schemes(pineapple_project, pod_generator_result.projects_by_pod_targets[pineapple_project], pod_generator_result)
+
+              scheme_path = Xcodeproj::XCScheme.shared_data_dir(pineapple_project.path) + 'PineappleLib-iOS.xcscheme'
+              scheme_path.should.exist?
+              test_scheme_path = Xcodeproj::XCScheme.shared_data_dir(pineapple_project.path) + 'PineappleLib-iOS-Unit-Tests.xcscheme'
+              test_scheme = Xcodeproj::XCScheme.new(test_scheme_path)
+              test_scheme.launch_action.macro_expansions.should.not.be.empty?
+
+              host_scheme_path = Xcodeproj::XCScheme.shared_data_dir(pineapple_project.path) + 'PineappleLib-iOS-App.xcscheme'
+              host_scheme = Xcodeproj::XCScheme.new(host_scheme_path)
+              native_target_uuids = @pineapple_spec.test_specs.
+                  map { |test_spec| pod_generator_result.native_target_for_spec(test_spec).uuid }.
+                  sort
+              host_scheme.test_action.testables.flat_map { |t| t.buildable_references.map(&:target_uuid) }.sort.should == native_target_uuids
             end
 
             it 'allows opting out' do
               @generator.installation_options.
                   stubs(:share_schemes_for_development_pods).
                   returns(false)
+              pod_generator_result = @generator.generate!
+              @generator.sandbox.stubs(:development_pods).returns('BananaLib' => fixture('BananaLib'))
+
+              project = pod_generator_result.projects_by_pod_targets.keys.find { |p| p.project_name == 'BananaLib' }
 
               Xcodeproj::XCScheme.expects(:share_scheme).never
-              @generator.configure_schemes(nil, [])
+              @generator.configure_schemes(project, pod_generator_result.projects_by_pod_targets[project], pod_generator_result)
```

Comment:
turns out the "opt out" tests weren't really testing anything, because they were passing `[]` for the pod target array which prevented the code path from ever triggering

https://github.com/CocoaPods/CocoaPods/blob/ef91fa13b2c19fc1e84aa38e19e658e5cf84c212/lib/cocoapods/installer/xcode/pods_project_generator.rb#L85-L90

`XCScheme.share_scheme` will _never_ be called if `pod_targets` is empty

Reply:
lol nice